include/awkward/array/EmptyArray.h

```diff
@@ -18,40 +18,40 @@ namespace awkward {
-    virtual const std::shared_ptr<Content> getitem_at_nowrap(int64_t at) const;
-    virtual const std::shared_ptr<Content> getitem_range(int64_t start, int64_t stop) const;
-    virtual const std::shared_ptr<Content> getitem_range_nowrap(int64_t start, int64_t stop) const;
-    virtual const std::shared_ptr<Content> getitem_field(const std::string& key) const;
-    virtual const std::shared_ptr<Content> getitem_fields(const std::vector<std::string>& keys) const;
-    virtual const std::shared_ptr<Content> carry(const Index64& carry) const;
-    virtual const std::pair<int64_t, int64_t> minmax_depth() const;
-    virtual int64_t numfields() const;
-    virtual int64_t fieldindex(const std::string& key) const;
-    virtual const std::string key(int64_t fieldindex) const;
-    virtual bool haskey(const std::string& key) const;
-    virtual const std::vector<std::string> keyaliases(int64_t fieldindex) const;
-    virtual const std::vector<std::string> keyaliases(const std::string& key) const;
-    virtual const std::vector<std::string> keys() const;
+    const std::string classname() const override;
+    void setid() override;
+    void setid(const std::shared_ptr<Identity> id) override;
+    const std::string tostring_part(const std::string indent, const std::string pre, const std::string post) const override;
+    void tojson_part(ToJson& builder) const override;
+    const std::shared_ptr<Type> innertype(bool bare) const override;
+    void settype_part(const std::shared_ptr<Type> type) override;
+    bool accepts(const std::shared_ptr<Type> type) override;
+    int64_t length() const override;
+    const std::shared_ptr<Content> shallow_copy() const override;
+    void check_for_iteration() const override;
+    const std::shared_ptr<Content> getitem_nothing() const override;
+    const std::shared_ptr<Content> getitem_at(int64_t at) const override;
+    const std::shared_ptr<Content> getitem_at_nowrap(int64_t at) const override;
+    const std::shared_ptr<Content> getitem_range(int64_t start, int64_t stop) const override;
+    const std::shared_ptr<Content> getitem_range_nowrap(int64_t start, int64_t stop) const override;
+    const std::shared_ptr<Content> getitem_field(const std::string& key) const override;
+    const std::shared_ptr<Content> getitem_fields(const std::vector<std::string>& keys) const override;
+    const std::shared_ptr<Content> carry(const Index64& carry) const override;
+    const std::pair<int64_t, int64_t> minmax_depth() const override;
+    int64_t numfields() const override;
+    int64_t fieldindex(const std::string& key) const override;
+    const std::string key(int64_t fieldindex) const override;
+    bool haskey(const std::string& key) const override;
+    const std::vector<std::string> keyaliases(int64_t fieldindex) const override;
+    const std::vector<std::string> keyaliases(const std::string& key) const override;
+    const std::vector<std::string> keys() const override;
```

Comment:
Replacing `virtual` with `override`: okay. I didn't know about that, but it provides some additional safety.